evalml/pipelines/component_graph.py

```diff
@@ -562,6 +564,39 @@ def _get_feature_provenance(self, input_feature_names):
             if len(children)
         }
 
+    def get_component_input_logical_types(self, component_name):
+        """Get the logical types that are passed to the given component.
+
+        Args:
+            component_name (str): Name of component in the graph
+
+        Returns:
+            Dict - Mapping feature name to logical type instance.
+
+        Raises:
+            ValueError: If the component is not in the graph.
+            ValueError: If the component graph as not been fitted
+        """
+        if not self._feature_logical_types:
+            raise ValueError("Component Graph has not been fit.")
+        if component_name not in self._feature_logical_types:
+            raise ValueError(f"Component {component_name} is not in the graph")
+
+        return self._feature_logical_types[component_name]
+
+    @property
+    def last_component_input_logical_types(self):
+        """Get the logical types that are passed to the last component in the pipeline.
+
+        Returns:
+            Dict - Mapping feature name to logical type instance.
+
+        Raises:
+            ValueError: If the component is not in the graph.
```

Comment:
A hypernit - I'm not sure what the convention is if a function itself does not directly raise an exception.  I feel like we can probably remove the Raises from here.  If you take it to a logical extreme, you could include tons of nested exceptions if you don't limit it to just the immediate scope of the function.  Feel free to just merge though, this is not a hill I will die on...or will I?